### PR TITLE
feat(share/discovery): persist discovered peers

### LINKED ISSUE #2208: share/discovery: persist discovered peers
Discovered peers, if not discarded are proven to be good, there is no need to rediscover peers on restart.

### PR DESCRIPTION
**Pull request by Wondertan**
Closes #2208 

- [ ] Integrate option
- [ ] Load peers into peer manager

### CODE REVIEW

**Comment by distractedm1nd** [comment]
I'm not convinced this is necessary outside of the context of saving peers for bootstrapper usage

**Comment by Wondertan** [comment]
> I'm not convinced this is necessary outside of the context of saving peers for bootstrapper usage, as I have never seen issues finding peers very quickly once we join the network. In fact, until other errors are fixed, this will probably give the disadvantage that we save bad peers (which is why restarting helps so often)

Think of this in conjunction with Discard, which filters unhealthy peers

File: share/availability/discovery/discovery_test.go [hunk 1]
```diff
@@ -109,6 +162,18 @@ func (t *testnet) discovery(opts ...Option) *Discovery {
 	return disc
 }
 
+func (t *testnet) borkenDiscovery(opts ...Option) *Discovery {
```
Comment (vgonkivs) [inline share/availability/discovery/discovery_test.go:165]: typo `brokenDiscovery`

Comment (Wondertan) [reply share/availability/discovery/discovery_test.go:165]: It was intentional 😈

**Pull request closed without merging**